### DISCUSSION TITLE
Duplicates on external library

### DISCUSSION THREAD

**Discussion by DdHms** [immich-app/immich#11798] [Q&A] [answered]
Hi, 
Love the project and hope to take part myself soon. 
I wanted to ask two question about duplicate detection: 
1. What is the duplication detection based on? Hash? Filename? Size? Is it near duplicate detection or exact duplicate detection. 
2. How would you suggest to remove duplicates in an external media volume? I guess if it is mounted as in the instructions read only (ro) it will not remove the original files. Should I temporarily mount with write permissions?

**Comment by C-Otto** [comment]
Some partial answers. Wildly different dimensions (WhatsApp vs. 10+ megapixel) still trigger duplicates. Smaller differences also count as duplicates. If you mount your external library RO, deletion (from trash) silently fails (with an error in the logs). If you mount RW, the files are deleted from the external library (when emptying the trash).

**Comment by gwlg-mjh** [comment, marked as answer]
From the settings page, it's under Machine Learning.  It looks like it uses the Smart Search vector embedding of each photo and then does a similarity search on that.  The default distance is 0.01 which is quite close but you can change that yourself if you want it to be even more similar or open out to be a bit more fuzzy.

I've had some photos from burst shots that have come up as duplicates because they are so, so similar but not quite the same, e.g. eyes looking slightly different direction.  Personally I'm happy with how that bit works.

I also think that the file hash is used to prevent exact duplicate files to be uploaded.  If you've uploaded some files and try and do it again they get hashed on the browser/app and don't even make it to the server, but this doesn't work on External libraries I don't think which is a big shame, and a feature I would gladly like to see added.
